File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/abstract_factory/lab/Lettuce.java
// Lettuce.java

public abstract class Lettuce
{
	public Lettuce(double aPrice)
	{
		myPrice = aPrice;
	}

	public Lettuce(Lettuce aLettuce)
	{
		myPrice = aLettuce.myPrice;
	}

	public abstract Lettuce clone();

	private double myPrice;
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_matlab_labs_and_solutions/prototype/solution/Lettuce.m
% Lettuce.m

classdef (Abstract) Lettuce < handle
	properties
		Price % int
	end

	methods
		% constructor:
		function theLettuce = Lettuce(aLettuceOrAPrice)
			if nargin == 1
				if isa(aLettuceOrAPrice, 'Lettuce')
					% Lettuce(aLettuce):
					theLettuce.Price = aLettuceOrAPrice.Price;
				else
					% Lettuce(aPrice):
					theLettuce.Price = aLettuceOrAPrice;
				end
			else
				error('Invalid number of input parameters.');
			end
		end
	end

	methods (Abstract)
		result = clone(theLettuce)
	end
end
